package/com.unity.formats.usd/Tests/Editor/EditorTests.cs

```diff
@@ -67,6 +75,29 @@ public void LoadAsFbxCompareMeshTransforms()
             Assert.AreApproximatelyEqual(usdMeshTr.localRotation.y,fbxMeshTr.localRotation.y);
             Assert.AreApproximatelyEqual(usdMeshTr.localRotation.z,fbxMeshTr.localRotation.z);
         }
+
+        [Test]
+        public void TestLeftHandedUsdImport()
+        {
+            // Compare import of Left handed USD to right handed USD
+            var leftHandedUsdRoot = LoadUSD(leftHandedUsdGUID);
+            Assert.IsNotNull(leftHandedUsdRoot);
+
+            // check that the mesh does not match the right handed one
+            var usdCube = usdRoot.transform.Find("group2/group1/pCube1");
+            Assert.IsNotNull(usdCube);
+            var leftHandedUsdCube = leftHandedUsdRoot.transform.Find("group2/group1/pCube1");
+            Assert.IsNotNull(leftHandedUsdCube);
+
+            var cubeMesh = usdCube.GetComponent<MeshFilter>().sharedMesh;
+            var leftHandedCubeMesh = leftHandedUsdCube.GetComponent<MeshFilter>().sharedMesh;
+            
+            // Since the two files are different handedness, the vertices, triangles, and normals
+            // will be different.
+            NUnit.Framework.Assert.That(cubeMesh.vertices, Is.Not.EqualTo(leftHandedCubeMesh.vertices));
+            NUnit.Framework.Assert.That(cubeMesh.triangles, Is.Not.EqualTo(leftHandedCubeMesh.triangles));
+            NUnit.Framework.Assert.That(cubeMesh.normals, Is.Not.EqualTo(leftHandedCubeMesh.normals));
```

Comment:
Is it enough to compare the data of two import results? Shouldn't we also test the generated mesh against a baked version to ensure the result we get is correct?

Reply:
This is actually a really good point -- especially with a symmetrical mesh like a Cube, you might even be getting identical results, but the floating point error alone might make them look different.

Does `Is.Not.EqualTo` use a fuzzy/epsilon comparison? If so, is the epsilon small enough to work here?

Reply:
I do like NOT using a baked/golden result though, because it makes the test more robust to other changes and avoids needing to ratchet a baseline.

Reply:
It should call Equals(), however, looking at the [Vector3 documentation](https://docs.unity3d.com/ScriptReference/Vector3.Equals.html) it looks like == is safer than Equals for approximate comparison, so I'll change this to be more explicit.